fog/view/enclave/impl/src/lib.rs

```diff
@@ -39,6 +42,13 @@ where
 
     /// Logger object
     logger: Logger,
+
+    /// Encrypts a QueryRequest for each individual Fog View Store.
+    /// TODO: Use a BTreeMap<FogViewShardLoadBalancerID,
+    /// BTreeMap<FogViewStoreId, Ready<...>>>  when implement the cursoring
+    /// optimization. For right now, it's fine to leave as a Vec because a
+    /// follow up PR will implement cursoring.
+    store_encryptors: Mutex<Vec<Ready<Aes256Gcm>>>,
```

Comment:
Would this need a Mutex? These carry the the ciphers, but from my read - I don't think the underlying data behind the cipher config changes much and there's not really a reason to lock it?

Reply:
Also, would different FOG View Stores be using different AES implementations?

Reply:
This needs to be a Mutex because the Cipher's `encrypt` method is mutable, and we can't get a mutable reference to `self` because the enclave can't provide this. Therefore we have to use a`Mutex` to enable interior mutability, which allows us to mutate a struct without having a mutable reference.